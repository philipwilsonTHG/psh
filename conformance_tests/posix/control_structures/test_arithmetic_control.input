# Test arithmetic in control flow
# PSH supports arithmetic commands and C-style loops (v0.31.0, v0.32.0 features)

# Test arithmetic conditions in if statements
x=10
y=5
if ((x > y)); then
    echo "$x is greater than $y"
fi

if ((x == 10)); then
    echo "x equals 10"
fi

if ((y < 3)); then
    echo "This should not print"
else
    echo "y is not less than 3"
fi

# Test arithmetic in while loops
echo "Countdown with arithmetic while:"
count=5
while ((count > 0)); do
    echo "Count: $count"
    ((count--))
done

# Test C-style for loops
echo "C-style for loop:"
for ((i = 1; i <= 3; i++)); do
    echo "Iteration: $i"
done

# Test C-style for with arithmetic expressions
echo "C-style for with complex arithmetic:"
for ((start = 2, end = 8; start < end; start += 2)); do
    echo "Value: $start"
done

# Test nested arithmetic control
echo "Nested arithmetic control:"
for ((outer = 1; outer <= 2; outer++)); do
    echo "Outer: $outer"
    inner=1
    while ((inner <= 2)); do
        echo "  Inner: $inner"
        if ((outer == 1 && inner == 2)); then
            echo "    Special case"
        fi
        ((inner++))
    done
done

# Test arithmetic command as statement
echo "Arithmetic command results:"
((result = 15 + 25))
echo "15 + 25 = $result"

((0))
echo "Exit status of ((0)): $?"

((1))
echo "Exit status of ((1)): $?"