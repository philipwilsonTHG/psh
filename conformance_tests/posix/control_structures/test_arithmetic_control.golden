10 is greater than 5
x equals 10
y is not less than 3
Countdown with arithmetic while:
Count: 5
Count: 4
Count: 3
Count: 2
Count: 1
C-style for loop:
Iteration: 1
Iteration: 2
Iteration: 3
C-style for with complex arithmetic:
Value: 2
Value: 4
Value: 6
Nested arithmetic control:
Outer: 1
  Inner: 1
  Inner: 2
    Special case
Outer: 2
  Inner: 1
  Inner: 2
Arithmetic command results:
15 + 25 = 40
Exit status of ((0)): 1
Exit status of ((1)): 0
